dask_ml/preprocessing/data.py

```diff
@@ -915,3 +916,43 @@ def inverse_transform(self, X):
             X[col] = series
 
         return X
+
+
+class PolynomialFeatures(skdata.PolynomialFeatures):
+
+    __doc__ = skdata.PolynomialFeatures.__doc__
+
+    def fit(self, X, y=None):
+
+        self._transformer = skdata.PolynomialFeatures()
+        X_sample = X
+        if isinstance(X, dd.DataFrame):
+            X_sample = X._meta_nonempty
+        if isinstance(X, da.Array):
+            X_sample = np.ones((1, X.shape[1]), dtype=X.dtype)
+
+        # pandas dataframe treated by sklearn and returns np.array
+        self._transformer.fit(X_sample)
+        copy_learned_attributes(self._transformer, self)
+        return self
+
+    def transform(self, X, y=None):
+
+        if isinstance(X, da.Array):
+            n_cols = len(self._transformer.get_feature_names())
+            # might raise Error if columns are separated by chunks
```

Comment:
Can you explain this error to me? Is the issue something like

```python
X = da.random.uniform(size=(100, 10), chunks=(25, 5))`
```

i.e. we have two blocks on the features?

Typically, we use dask_ml.utils.check_array to handle this. There's a `accept_multiple_blocks` keyword.

I think that we *want* to raise an exception in this case, since the transform would need to be different for the different blocks.

Reply:
Exactly. Sorry for not directly raising an error here. Thanks for the hint on the accept_multiple_blocks.

Reply:
Okay, added `check_array`. Btw, should I also use check_array for the dask dataframe?